src/mapgen/mapgen_valleys.cpp

```diff
@@ -477,17 +477,16 @@ int MapgenValleys::generateTerrain()
 		if (surface_y > surface_max_y)
 			surface_max_y = std::ceil(surface_y);
 
-		if (humid_rivers) {
-			// Derive heat from (base) altitude. This will be most correct
-			// at rivers, since other surface heights may vary below.
-			if (use_altitude_chill && (surface_y > 0.0f || river_y > 0.0f))
-				t_heat -= alt_to_heat *
-					std::fmax(surface_y, river_y) / altitude_chill;
-
-			// If humidity is low or heat is high, lower the water table
+		// Optionally vary river depth according to heat and humidity
+		if (!(spflags & MGVALLEYS_FIXED_RIVER_DEPTH)) {
+			float heat = ((spflags & MGVALLEYS_ALT_CHILL) &&
+				(surface_y > 0.0f || river_y > 0.0f)) ?
```

Comment:
An if-statement as before would make it more readable - maybe also with a comment line if appropriate.

Reply:
Not sure, seems fairly readable to me.

Reply:
Thisis the sort of code where hmmmm would have said to me 'use a tertiary operator'.

Reply:
However, if you insist, i will change it, i'm not too bothered :) If it helps me get an approval i will certainly change it.